post.md

```diff
@@ -0,0 +1,411 @@
+---
+post_title: Basics of x86 assembly (AT&T syntax)
+author: Bartek Pastudzki
+post_excerpt: "This article is meant to preasent the most important ideas behind x86 assembly just to show how does it work, what are its limitations."
+layout: post
+published: true
+---
+
+Nowadays, application of programming in assembly language is
+very small. Writing production code in assembly may be found
+in the most demanding tasks in embedded. Even in modern firmware
+(coreboot, EDK2) most of the code is written in C. Honesly,
+that's quite undestandable, assembly code isn't easy to read
+nor write and some state that it's no longer needed to be known.
+In my opinion, that's far from truth.
+
+Not only it's heavily used in reverse engineering, very important
+especially in software/firmware security, its design impact more
+or less code in virtualy any programming language. Let's set an
+example. Consider those two lines of C code:
```

Comment:
Not only it's heavily used in reverse engineering, very important especially in software/firmware security, its design impacts, more or less, code in virtually any programming language. Let's set an example. Consider those two lines of C code: